File: Hash Map & Set/Medium/2352. Equal Row and Column Pairs/equal_row_and_column_pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        map<vector<int>,int> freq;
        int n=grid.size();
        int count=0;

        for(auto i: grid)
        {
            freq[i]++;
        }

        for(int j=0; j<n; j++)
        {
            vector<int> helper;
            for(int i=0; i<n; i++)
            {
                helper.push_back(grid[i][j]);
            }

            count+=freq[helper];
        }

        return count;
    }
};

File: Hash Map & Set/Medium/2352. Equal Row and Column Pairs/equal_row_and_column_pairs.py
class Solution:
    def equalPairs(self, grid: list[list[int]]) -> int:
        freq = defaultdict(int)
        n = len(grid)
        count = 0

        # Count frequency of each row (as a tuple)
        for row in grid:
            freq[tuple(row)] += 1

        # For each column, build the tuple and check frequency
        for j in range(n):
            col = tuple(grid[i][j] for i in range(n))
            count += freq[col]

        return count
